### PR TITLE
Add Money unit test

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add unit test
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Travis is green

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15431)

### CODE REVIEW

File: tests/Unit/Core/Domain/ValueObject/MoneyTest.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Money;
+
+class MoneyTest extends TestCase
+{
+    public function testCreateWithValidValue()
+    {
+        $money = new Money(new Number('100'), new CurrencyId(10));
+
+        $this->assertEquals(100, $money->getAmount());
```
Comment (PierreRambaud) [inline tests/Unit/Core/Domain/ValueObject/MoneyTest.php]: getAmount returns a Number, not a Integer ;)
```suggestion
        $this->assertEquals('100', (string) $money->getAmount());
```

Comment (matks) [reply tests/Unit/Core/Domain/ValueObject/MoneyTest.php]: I knew it 😆 I thought to myself "do I need to run such a simple test before the PR ? naaaaah I'm fine"

I got punished by fate 😉

**Comment by matks** [comment]
@PierreRambaud fixed

By the way I'm still unsure whether a `Money` object should contain a `$currencyId` property or a `$isoCode` property 🤔 

The currencyId is needed anyways in order to be able to perform database-related operations but could be retrieved later. Using `$isoCode` would make the object simpler to create. Wdyt ?

**Comment by PierreRambaud** [comment]
> @PierreRambaud fixed
> 
> By the way I'm still unsure whether a `Money` object should contain a `$currencyId` property or a `$isoCode` property 
> 
> The currencyId is needed anyways in order to be able to perform database-related operations but could be retrieved later. Using `$isoCode` would make the object simpler to create. Wdyt ?

I'm not the best one to answer you :sweat_smile: 
But I think to declare a `Money` you need a `Currency` and also an `IsoCode` because money are based on a currency and where it's used :thinking: 
PIng @eternoendless

File: tests/Unit/Core/Domain/ValueObject/MoneyTest.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+        $this->assertEquals('100', (string) $money->getAmount());
+    }
+
+    public function testCreateWithBadValue()
+    {
+        $this->expectException(DomainConstraintException::class);
+        $this->expectExceptionCode(DomainConstraintException::INVALID_MONEY_AMOUNT);
+
+        $money = new Money(new Number('-1'), new CurrencyId(10));
```
Comment (eternoendless) [inline tests/Unit/Core/Domain/ValueObject/MoneyTest.php]: O_o, why would a negative money amount be invalid?

Comment (matks) [reply tests/Unit/Core/Domain/ValueObject/MoneyTest.php]: That's why this test checks an exception is thrown 😉

Comment (eternoendless) [reply tests/Unit/Core/Domain/ValueObject/MoneyTest.php]: The problem is that a negative money amount is conceptually valid (eg. a discount, a reimbursement)

Comment (matks) [reply tests/Unit/Core/Domain/ValueObject/MoneyTest.php]: Depends on you consider it. You can consider a discount has the informations: "amount", "10", "action": "deduce". The VO Money has been introduced to handle CartRule informations, where the amount and he action are too separate things. But we can remove this validation rule very easily.

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — COMMENTED** [review]
@matks Some PrettyCI to fix and a second review for this technical PR.

**Comment by matks** [comment]
@Progi1984 fixed 😄

**Review by Progi1984 — APPROVED** [review]

**Comment by matks** [comment]
Thanks for review !

**Pull request merged**